slf4j/src/main/scala/zio/logging/slf4j/SLF4J.scala

```diff
@@ -105,7 +105,11 @@ object SLF4J {
        */
       override def appendCause(cause: Cause[Any]): Unit = {
         if (!cause.isEmpty) {
-          throwable = FiberFailure(cause)
+          cause match {
```

Comment:
Matching on a cause directly is not reliable because of nodes like or / and / stackless /e tc.

Instead, if you are trying to just "get the most important throwable", then do something like `squash` or `defects` or perhaps: `(cause.errors.collect { case t : Throwable } ++ cause.defects).head` (or its equivalent). 

Because you want to prefer an error before you look into the defects, and you want to be robust in the presence of intermediate / metadata nodes.

Reply:
@jdegoes 

I changed that to
```scala
        if (!cause.isEmpty) {
          val maybeThrowable = (cause.failures.collect { case t: Throwable => t } ++ cause.defects).headOption
          maybeThrowable match {
            case Some(t) => throwable = t
            case None    => throwable = FiberFailure(cause)
          }
        }
```

can you check it? Thank you